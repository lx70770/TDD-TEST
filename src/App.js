import React from "react";

function App() {
  return (
    <div className="app-container" name="lixiang" data-jest="container">
      aaa
    </div>
  );
}

export default App;
