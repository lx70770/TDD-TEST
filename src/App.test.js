import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import App from "./App";

Enzyme.configure({ adapter: new Adapter() });

describe("测试enzyme的shallow", () => {
  const wrapper = shallow(<App />);
  const container = wrapper.find("[data-jest='container']");

  it("使用Enzyme测试DOM元素", () => {
    console.log(wrapper.debug());
    expect(container.length).toBe(1);
    expect(container.prop("name")).toBe("lixiang");
  });

  it("使用jest-enzyme方便测试DOM", () => {
    expect(container).toExist();
    expect(container).toHaveProp("name", "lixiang");
  });
});

describe("测试enzyme的mount", () => {
  const wrapper = mount(<App />);

  // 某些场景是不能修改DOM元素的 可以用快照测试
  it("使用Enzyme测试快照", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
