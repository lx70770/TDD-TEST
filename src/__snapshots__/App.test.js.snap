// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试enzyme的mount 使用Enzyme测试快照 1`] = `
<App>
  <div
    className="app-container"
    data-jest="container"
    name="lixiang"
  >
    aaa
  </div>
</App>
`;
